src/file_drop/mod.rs

```diff
@@ -0,0 +1,147 @@
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub(crate) use linux::*;
+#[cfg(target_os = "macos")]
+mod macos;
+#[cfg(target_os = "macos")]
+pub(crate) use macos::*;
+#[cfg(target_os = "windows")]
+mod win;
+#[cfg(target_os = "windows")]
+pub(crate) use win::*;
+
+use core::fmt;
+use std::{cell::Cell, path::PathBuf, sync::Arc};
+
+#[derive(Debug, Serialize, Clone)]
+/// The status of a file drop event.
+pub enum FileDropStatus {
+    /// The file(s) have been dragged onto the window, but have not been dropped yet.
+    Hovered(Vec<PathBuf>),
+
+    /// The file(s) have been dropped onto the window.
+    Dropped(Vec<PathBuf>),
+
+    /// The file(s) drop was aborted.
+    Cancelled(Vec<PathBuf>),
+}
+
+// This needs to be defined because internally the respective events do not always yield a PathBuf.
+// We can easily remember what was cancelled though, as Hovered and Dropped events will always yield a PathBuf which we will save ourselves for later reference.
+pub(crate) enum FileDropEvent {
+    Hovered,
+    Dropped,
+    Cancelled
+}
+
+#[derive(Clone)]
+pub struct FileDropHandler {
+    f: Arc<Box<dyn Fn(FileDropStatus) -> bool + Send + Sync + 'static>>
+}
+impl FileDropHandler {
+    /// Initializes a new file drop handler.
+    ///
+    /// Example: FileDropHandler:new(|status: FileDropStatus| ...)
+    /// 
+    /// ### Blocking OS Default Behavior
+    /// Return `true` in the callback to block the OS' default behavior of handling a file drop.
+    ///
+    /// Note, that if you do block this behavior, it won't be possible to drop files on `<input type="file">` forms.
+    /// Also note, that it's not possible to manually set the value of a `<input type="file">` via JavaScript for security reasons.
```

Comment:
Just a question: does an `eval` from the rust side of the webview suffer the same limitation?